api_app/analyzers_manager/observable_analyzers/ipqs.py

```diff
@@ -0,0 +1,163 @@
+        payload = {
+            "strictness": self.ip_strictness,
+            "allow_public_access_points": str(self.allow_public_access_points).lower(),
+            "fast": str(self.ip_fast).lower(),
+            "lighter_penalties": str(self.lighter_penalties).lower(),
+            "mobile": str(self.mobile).lower(),
+            "transaction_strictness": self.transaction_strictness,
+        }
+        if self.user_agent:
+            payload["user_agent"] = self.user_agent
+        if self.user_language:
+            payload["user_language"] = self.user_language
+        return payload
+
+    def _get_email_payload(self):
+        return {
+            "fast": str(self.email_fast).lower(),
+            "timeout": self.email_timeout,
+            "suggest_domain": str(self.suggest_domain).lower(),
+            "strictness": self.email_strictness,
+            "abuse_strictness": self.abuse_strictness,
+        }
+
+    def _get_phone_payload(self):
+        return {
+            "strictness": self.phone_strictness,
+            "country": [self.country],
+            "enhanced_line_check": str(self.enhanced_line_check).lower(),
+            "enhanced_name_check": str(self.enhanced_name_check).lower(),
+        }
+
+    def _get_calling_endpoint(self):
+        if re.match(IP_REG, self.observable_name) or re.match(
+            IPv6_REG, self.observable_name
+        ):
+            return self.IP_ENDPOINT, self._get_ip_payload()
+        elif re.match(DOMAIN_REG, self.observable_name) or re.match(
+            URL_REG, self.observable_name
+        ):
+            return self.URL_ENDPOINT, self._get_url_payload()
+        elif re.match(EMAIL_REG, self.observable_name):
+            return self.EMAIL_ENDPOINT, self._get_email_payload()
+        elif re.match(PHONE_REG, self.observable_name):
+            return self.PHONE_ENDPOINT, self._get_phone_payload()
+        else:
+            return None, None
+
+    def run(self):
+        calling_endpoint, payload = self._get_calling_endpoint()
+        ipqs_headers = {"IPQS-KEY": self._ipqs_api_key}
+
+        try:
+            if calling_endpoint and payload is not None:
+                response = requests.get(
+                    calling_endpoint + self.observable_name,
+                    headers=ipqs_headers,
+                    params=payload,
+                )
+                response.raise_for_status()
+                result = response.json()
+                return result
+            else:
+                logger.warning("Invalid or unsupported observable type")
+                raise AnalyzerRunException("Invalid or unsupported observable type")
+        except requests.RequestException as e:
+            raise AnalyzerRunException(e)
+
+    @classmethod
+    def _monkeypatch(cls):
+        patches = [
+            if_mock_connections(
+                patch(
+                    "requests.get",
+                    return_value=MockUpResponse({}, 200),
```

Comment:
can you please add here an example of a possible output of the tool? ty

Reply:
> can you please add here an example of a possible output of the tool? ty

Thanks @mlodic , please check now